### ISSUE TITLE
workgroup4

### ISSUE THREAD

**Issue by alexanderquispe** [d2cml-ai/CausalAI-Course#247]
# Orthogonal Learning
1. You need to translate the notebook-experiment-on-orthogonal-learning into Julia. 
Here are the scripts in [Python](https://github.com/alexanderquispe/ECO224/blob/main/Labs/Python_Notebooks/Lab4/python_notebook_experiment_on_orthogonal_learning.ipynb) and [R](https://github.com/alexanderquispe/ECO224/blob/main/Labs/R_Notebooks/Lab4/r-notebook-experiment-on-orthogonal-learning.ipynb). 

2. Based on those scripts you will replicate Simulation Design 1 three times using different numbers of trials. So you have to use
- The equation of Y for the three cases now is: Y = 5\*D + beta\*X + rnorm(n)
-  B = 50 , B = 100, B = 1000. 
- Explain what you observe in the distribution of the histograms. Where the orthogonal and Naive should be centered? 
- Why does this happen? Give an econometric explanation of this. 
- Label your axes and legends correctly in your figures. 
- The analysis should be done in Python , R and Julia

# Double Lasso - heterogenous wage effects

1. Complete the third stage of partialling out in Julia's code - heterogeneous effects. Review the last half hour of the recording (PD)  where the solution was shown.

# Double Lasso - Testing the Convergence Hypothesis

You have to create your analysis based on the scripts **pm2_notebook_jannis** in Python, R and Julia.
1. Explain what is the Double Lasso Approach in a markdown cell. Use equations for a better explanation. 
3. Replicate the results using the next approaches:

- OLS
- Double LASSO using cross Validation ([GLMNET ](https://glmnet.stanford.edu/articles/glmnet.html)in R and [Sklearn ](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LassoCV.html)in Python )
- Double Lasso using theoretical Lambda (HDM package both in R and Python)
- Double Lasso using **method="partialling out"**, this is a direct way to get results from the HDM package, check the Python script for an example, and also you have to read the [documentation](https://arxiv.org/pdf/1603.01700.pdf)  for more details. 
- Plot the main coefficient and its confidence interval of the convergence hypothesis from these 4 cases in one figure. 

# Neyman Orthogonality Proof
Using Markdown cells You have to prove the Neyman Orthogonality Condition. Please, first **read** this [Neyman_orthogonality_Proof](https://github.com/alexanderquispe/ECO224/blob/main/Lectures/Lecture_5/Neyman%20Orthogonality_proof.pdf) as tutorial.
Please, read the last page of this pdf where more details are about how to make a correct proof.            

# God and Bad Controls
You have to create a tutorial about good and bad controls. 
Please **read** this [tutorial](https://github.com/alexanderquispe/ECO224/tree/main/Labs/Python_Notebooks/Lab5) and create one for the 18 cases shown in [Cinelli et al 2022](https://github.com/alexanderquispe/ECO224/blob/main/Reports/week_5/crash_course_bad_controls.pdf).
Follow the next steps:
1. For each case create the associated DAG. You will need to use the next libraries for each programming language. 

- In R, use [dagitty](http://dagitty.net/primer/)
- in Python use [dowhy](https://microsoft.github.io/dowhy/example_notebooks/tutorial-causalinference-machinelearning-using-dowhy-econml.html) or [causalgraphicalmodels](https://github.com/ijmbarr/causalgraphicalmodels)
- In Julia use [StructuralCausalModels](https://github.com/StatisticalRethinkingJulia/StructuralCausalModels.jl) you can also check this [tutorial](https://github.com/alexanderquispe/14.38_Causal_ML/blob/main/Julia_Notebooks/Notebook_DAGitty.ipynb) or you can use [causalinference.jl](https://mschauer.github.io/CausalInference.jl/latest/examples/pc_basic_examples/). Choose the one you prefer. 

2. The effect from **X to Y should be equal to 1**. You can choose the effect from other covariates into X, but try to keep similar effects in the other regressions. 

**Deadline: 12/05/22 at 18:00.**
